File: testClients/traderThief/AI.py
# -*- coding: iso-8859-1 -*-
#-*-python-*-
from BaseAI import BaseAI
from GameObject import *

class AI(BaseAI):
  """The class implementing gameplay logic."""
  @staticmethod
  def username():
    return "Shell AI"

  @staticmethod
  def password():
    return "password"

  def init(self):
    pass

  def end(self):
    pass

  def run(self):
      print "Units:" + `len(self.pirates)`
      myUnits = [i for i in self.pirates if i.getOwner() == self.playerID()]
      myShips = [i for i in self.ships if i.getOwner() == self.playerID()]
      myPorts = [i for i in self.ports if i.getOwner() == self.playerID()]
      if self.players[self.playerID()].getGold() > self.shipCost() and len(myShips) == 0:
        myPorts[0].createShip()
      for i in range(0,int( self.players[self.playerID()].getGold() / self.pirateCost())):
        myPorts[0].createPirate()
      print len(myUnits)
      for i in myUnits:
        for j in self.pirates:
          if j.getOwner() != self.playerID:
            if abs(j.getX()-i.getX())+abs(j.getY()-i.getY()) == 1:
              if i.getAttacksLeft() > 0:
                i.attack(j)
                break
        if i.getGold() > 0:
          i.dropTreasure(i.getGold())
      for i in myShips:
        for j in self.ships:
          if j.getOwner() != self.playerID:
            if abs(j.getX()-i.getX())+abs(j.getY()-i.getY()) == 1:
              if i.getAttacksLeft() > 0:
                i.attack(j)
                break
      return 1
      
      

  def __init__(self, conn):
      BaseAI.__init__(self, conn)

File: testClients/gameOverCheck/AI.cpp
#include "AI.h"
#include "util.h"

AI::AI(Connection* conn) : BaseAI(conn) {}

const char* AI::username()
{
  return "Shell AI";
}

const char* AI::password()
{
  return "password";
}

//This function is run once, before your first turn.
void AI::init(){}

//This function is called each time it is your turn.
//Return true to end your turn, return false to ask the server for updated information.
bool AI::run()
{
  // TO CHANGE WIN CONDIITON, CHANGE THIS!
  int keepPirates=1;
  int keepShips=0;
  int keepGold=0;
  int countPirates=0,countShips=0,countGold=0;
  
  // find your port
  int myPort=0;
  while(ports[myPort].owner() != playerID())myPort++;

  // spend money until you are down to the desired gold
  while(players[playerID()].gold()- countGold > keepGold)
  {
    cout<<"Buying a pirate"<<endl;
    ports[myPort].createPirate();
    countGold+=pirateCost();
  }
  for(size_t i=0;i<pirates.size();i++)
  {
    if(pirates[i].owner()==playerID())
    {
      if(countPirates < keepPirates)
      {
        countPirates++;
      }
      else
      {
        cout<<"Killing a pirate"<<endl;
        pirates[i].attack(pirates[i]);
      }
    }
  }
  for(size_t i=0;i<ships.size();i++)
  {
    if(ships[i].owner()==playerID())
    {
      if(countShips < keepShips)
      {
        countShips++;
      }
      else
      {
        cout<<"Ship attacking itself"<<endl;
        ships[i].attack(ships[i]);
      }
    }
  }
  cout<<"Should have: "<<countPirates<<" "<<countShips<<" "<<players[playerID()].gold()-countGold<<endl;
  return true;
}

//This function is run once, after your last turn.
void AI::end(){}
